src/Designer/backend/src/Designer/Clients/Implementations/AzureSharedContentClient.cs

```diff
@@ -306,4 +309,53 @@ internal void SetCurrentVersion(List<string> versionPrefixes)
         int version = versions.Max();
         CurrentVersion = (version + 1).ToString();
     }
+
+    public async Task<List<string>> GetPublishedResourcesForOrg(string orgName, string path = "", CancellationToken cancellationToken = default)
+    {
+        string prefix = orgName + "/" + path;
+        return await GetPublishedResourcesWithPrefix(prefix, cancellationToken);
+    }
+
+    private async Task<List<string>> GetPublishedResourcesWithPrefix(string prefix, CancellationToken cancellationToken)
+    {
+        List<BlobItem> blobs = await GetBlobsWithPrefix(prefix, cancellationToken);
+        return blobs
+            .Select(b => RemovePrefixAndLeadingSlash(b.Name, prefix))
+            .ToList();
+    }
+
+    private async Task<List<BlobItem>> GetBlobsWithPrefix(string prefix, CancellationToken cancellationToken)
+    {
+        try
+        {
+            AsyncPageable<BlobItem> blobs = GetPageableBlobsWithPrefix(prefix, cancellationToken);
+            return await EnumerateBlobs(blobs, cancellationToken);
+        }
+        catch (Exception ex) when (ex is RequestFailedException or AggregateException)
+        {
+            _logger.LogError(ex, "Error fetching blobs with prefix {Prefix} in {Class}", prefix, nameof(AzureSharedContentClient));
+            throw new SharedContentRequestException($"Error fetching blobs with prefix {prefix}", ex);
+        }
+    }
+
+    private AsyncPageable<BlobItem> GetPageableBlobsWithPrefix(string prefix, CancellationToken cancellationToken)
+    {
+        BlobContainerClient containerClient = _blobContainerClientFactory.GetContainerClient();
+        return containerClient.GetBlobsAsync(BlobTraits.None, BlobStates.None, prefix, cancellationToken);
+    }
+
+    private static async Task<List<BlobItem>> EnumerateBlobs(
+        AsyncPageable<BlobItem> enumerableBlobs,
+        CancellationToken cancellationToken
+    )
+    {
+        List<BlobItem> blobItemList = [];
+        await foreach (BlobItem b in enumerableBlobs.WithCancellation(cancellationToken)) { blobItemList.Add(b); }
```

Comment:
Suggestion to make it multiline to be able to see it easier.
```suggestion
        await foreach (BlobItem b in enumerableBlobs.WithCancellation(cancellationToken))
        {
            blobItemList.Add(b);
        }}
```